strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -249,17 +278,22 @@ public void deleteMetadata(String storageId,
             else
             {
                 // This is at the artifact level
-                Path mavenMetadataPath = Files.list(artifactVersionPath.getParent())
-                                              .filter(p -> p.getFileName().toString().endsWith("maven-metadata.xml"))
-                                              .findFirst()
-                                              .orElse(null);
-
-                if (mavenMetadataPath != null)
+                try (Stream<Path> pathStream = Files.list(artifactVersionPath.getParent()))
```

Comment:
> Don't you think that closing unused resources is a good practice ? :P
hm.... I just mean that it's not so convenient, we can say that I just share my indignation :)